rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/jobs/local/JobExecutorLocal.groovy

```diff
@@ -161,14 +169,23 @@ class JobExecutorLocal implements JobExecutor {
   void cancelJob(String jobId) {
     log.info("Canceling job $jobId...")
 
-    // Terminate the process.
-    if (jobIdToHandlerMap[jobId]) {
-      jobIdToHandlerMap[jobId].watchdog.destroyProcess()
-    }
-
     // Remove the job from this rosco instance's handler map.
-    jobIdToHandlerMap.remove(jobId)
+    def canceledJob = jobIdToHandlerMap.remove(jobId)
+
+    // Terminate the process.
+    canceledJob?.watchdog?.destroyProcess()
 
     // The next polling interval will be unable to retrieve the job status and will mark the bake as canceled.
   }
+
+  @PostConstruct
+  void initializeMetrics() {
+    registry.gauge(registry.createId("bakes.local").withTag("active", "true"), jobIdToHandlerMap, new ToDoubleFunction<Map>() {
+
```

Comment:
I used the double because that's what the `registry.gauge()` method expected. I couldn't find a non-double version.